infrastructure/src/main/java/org/corfudb/infrastructure/datastore/DataStore.java

```diff
@@ -223,7 +233,7 @@ public synchronized void delete(@Nonnull String key,
     }
 
     @Override
-    public synchronized <T> void delete(KvRecord<T> key) {
+    public <T> void delete(KvRecord<T> key) {
         cache.invalidate(key.getFullKeyName());
     }
```

Comment:
Calling delete and not doing anything is a little weird. I would rather just delete the method from the interface. It's not used anyways.

Reply:
Delete will remove the in-memory cache entry, but not remove the underlying file. This is being used in tests.

Reply:
It's only used in the test. I think `CacheWriter` is not needed here, you can probably use a loading cache. 
If you really want to keep it then maybe call it `invalidate.` 

```java
/**
   * Deletes the value corresponding to the {@code key} from the external resource. The cache will
   * communicate a delete when the entry is explicitly removed or evicted.
   *
   * @param key the non-null key whose value was removed
   * @param value the value associated with {@code key}, or {@code null} if collected
   * @param cause the reason for which the entry was removed
   * @throws RuntimeException or Error, in which case the mapping is unchanged
   */
  void delete(@Nonnull K key, @Nullable V value, @Nonnull RemovalCause cause);

```

Reply:
@Maithem I think the reason that we need a CacheWriter is because the write to cache and disk needs to be atomic. And since the interface is a generic kv datastore instead of a cache, maybe it's better to still call it delete?

Reply:
I see. You can just use the cache as a concurrent hash map if you need per-key atomicity. My only point is why keep it if you dont need it.

Reply:
Yea, right now we need this interface for tests, and since it's a generic kv store interface, maybe later there will be another impl that actually does delete for purposes other than tests.